pkg/api/pod/util_test.go

```diff
@@ -1552,3 +1552,93 @@ func TestDropSubPathExpr(t *testing.T) {
 		}
 	}
 }
+
+func TestDropTopologySpreadConstraints(t *testing.T) {
+	podWithTopologySpreadConstraints := func() *api.Pod {
+		return &api.Pod{
+			Spec: api.PodSpec{
+				TopologySpreadConstraints: []api.TopologySpreadConstraint{
+					{
+						MaxSkew:           1,
+						TopologyKey:       "k8s.io/hostname",
+						WhenUnsatisfiable: api.DoNotSchedule,
+					},
+				},
+			},
+		}
+	}
+	podWithoutTopologySpreadConstraints := func() *api.Pod {
+		return &api.Pod{}
+	}
+
+	podInfo := []struct {
+		description    string
+		hasConstraints bool
+		pod            func() *api.Pod
+	}{
+		{
```

Comment:
Why don't we put a pod, whether the feature is enabled/disabled, and the expected pod in each test case and check the pod coming out of `dropDisabledFields` against the expected pod? I think that would simplify the test.

Reply:
This style is consistent with other tests in this file. I'm not 100% sure about the original intent.

This test is a 2(enabled or not) * 2(old spec with or without constraints) *2(new spec with or without constraints) matrix, I _guess_ it's b/c if we go with the {want, expected} style, we have to give 8 {want, expected} tuples, and the number would increase exponentially if we have other dimensions in the matrix.

Reply:
Hmm.. I'm following the style in this file. So maybe we can come up with a follow-up PR discussing and proposing a new style.

Reply:
I agree w/ @bsalamat, I don't see the purpose of the "hasConstraints" variable. We shouldn't copy a style if we don't understand its purpose :)

Reply:
I've removed the variable "hasConstraints" as it can be calculated. But remains the "matrix" testing style.